herd/AArch64Sem.ml

```diff
@@ -39,6 +39,7 @@ module Make
     let kvm = C.variant Variant.Kvm
     let is_branching = kvm && not (C.variant Variant.NoPteBranch)
     let pte2 = kvm && C.variant Variant.PTE2
+    let do_cu = C.variant Variant.ConstrainedUnpredictable
```

Comment:
I am wondering whether it would be better to instead raise a flag ContrainedUnpredictable and keep things simple. The problem is that there are multiple parts of the architecture that allow for ContrainedUnpredictable behaviors, not only LDRX/STRX. Also an given implementation makes a certain choice (e.g., Exclusives monitor on a cache-line granularity) which means that if the LDRX/STRX pair will succeed if some conditions are met and not unconditionally.

Reply:
The general idea is to avoid Contrained Unpredictable by default. The `-variant ContrainedUnpredictable` command line option instructs herd to take the opposite view to yield such behaviours, _as much as possible_. In the case of LDRX/STRX this means that store exclusive are possible, regardless of the reservation value (if any). I am not sure it is desirable to instruct herd about details such as granule size etc. Other Contrained Unpredictable are not handled yet and may be in the future. Hopefully, one variant will be enough...